// @vx/grid/lib/grids/GridRows.js
"use strict";

exports.__esModule = true;
exports.default = GridRows;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Line = _interopRequireDefault(require("@vx/shape/lib/shapes/Line"));

var _group = require("@vx/group");

var _point = require("@vx/point");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function GridRows(_ref) {
  var _ref$top = _ref.top,
      top = _ref$top === void 0 ? 0 : _ref$top,
      _ref$left = _ref.left,
      left = _ref$left === void 0 ? 0 : _ref$left,
      scale = _ref.scale,
      width = _ref.width,
      _ref$stroke = _ref.stroke,
      stroke = _ref$stroke === void 0 ? '#eaf0f6' : _ref$stroke,
      _ref$strokeWidth = _ref.strokeWidth,
      strokeWidth = _ref$strokeWidth === void 0 ? 1 : _ref$strokeWidth,
      strokeDasharray = _ref.strokeDasharray,
      className = _ref.className,
      _ref$numTicks = _ref.numTicks,
      numTicks = _ref$numTicks === void 0 ? 10 : _ref$numTicks,
      lineStyle = _ref.lineStyle,
      offset = _ref.offset,
      tickValues = _ref.tickValues,
      restProps = _objectWithoutPropertiesLoose(_ref, ["top", "left", "scale", "width", "stroke", "strokeWidth", "strokeDasharray", "className", "numTicks", "lineStyle", "offset", "tickValues"]);

  var ticks = tickValues || (scale.ticks ? scale.ticks(numTicks) : scale.domain());
  return _react.default.createElement(_group.Group, {
    className: (0, _classnames.default)('vx-rows', className),
    top: top,
    left: left
  }, ticks.map(function (d, i) {
    var y = offset ? (scale(d) || 0) + offset : scale(d) || 0;
    var fromPoint = new _point.Point({
      x: 0,
      y: y
    });
    var toPoint = new _point.Point({
      x: width,
      y: y
    });
    return _react.default.createElement(_Line.default, _extends({
      key: "row-line-" + d + "-" + i,
      from: fromPoint,
      to: toPoint,
      stroke: stroke,
      strokeWidth: strokeWidth,
      strokeDasharray: strokeDasharray,
      style: lineStyle
    }, restProps));
  }));
}

GridRows.propTypes = {
  width: _propTypes.default.number.isRequired
};
